From 821e53327847a5acdae57c1a905ebd95d4965489 Mon Sep 17 00:00:00 2001
From: Dorin Ionita <dorin.ionita@nxp.com>
Date: Tue, 19 Apr 2022 20:34:36 +0300
Subject: [PATCH 31/33] gmac: s32: Fixed GMAC clock failed message in U-Boot
 log.

When the hwconfig variable was not set, failed clock setting messages
were shown in the U-Boot log. This happened because the clock init
function was actually called both before and after checking the Serdes
configuration.
Fixed by removing the duplicate call before that check.

Issue: ALB-8364
Upstream-Status: Pending 

Signed-off-by: Dorin Ionita <dorin.ionita@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/dwc_eth_qos_s32cc.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/net/dwc_eth_qos_s32cc.c b/drivers/net/dwc_eth_qos_s32cc.c
index c2b153a023..f3e7c6e742 100644
--- a/drivers/net/dwc_eth_qos_s32cc.c
+++ b/drivers/net/dwc_eth_qos_s32cc.c
@@ -22,8 +22,8 @@
 #include <phy.h>
 #include <reset.h>
 #include <wait_bit.h>
-#include <asm/arch/clock.h>
 #include <asm/io.h>
+#include <asm/arch/clock.h>
 
 #include "board_common.h"
 #include "serdes_regs.h"
@@ -31,10 +31,10 @@
 
 #include <dm/platform_data/dwc_eth_qos_dm.h>
 
-#include "dwc_eth_qos.h"
 #include <s32gen1_clk_utils.h>
 #include <s32gen1_gmac_utils.h>
 #include <dm/device.h>
+#include "dwc_eth_qos.h"
 
 /* S32 SRC register for phyif selection */
 #define PHY_INTF_SEL_SGMII	0x01
@@ -204,7 +204,6 @@ static bool s32ccgmac_set_interface(struct udevice *dev, phy_interface_t mode)
 		return false;
 
 	setup_iomux_enet_gmac(dev, mode);
-	setup_clocks_enet_gmac(mode, dev);
 	s32cc->mac_intf = mode;
 
 	return true;
-- 
2.17.1

